py/server/deephaven/table.py

```diff
@@ -343,14 +343,37 @@ def coalesce(self) -> Table:
         return Table(j_table=self.j_table.coalesce())
 
     def snapshot(self) -> Table:
-        """ Returns a static snapshot table."""
+        """Returns a static snapshot table."""
         try:
             with auto_locking_ctx(self):
                 return Table(j_table=self.j_table.snapshot())
         except Exception as e:
             raise DHError(message="failed to create a snapshot.") from e
 
-    def snapshot_when(self, trigger_table: Table, initial: bool = False, incremental: bool = False, history: bool = False, stamp_cols: Union[str, List[str]] = None) -> Table:
+    def snapshot_when(self, trigger_table: Table, stamp_cols: Union[str, List[str]] = None, initial: bool = False, incremental: bool = False, history: bool = False) -> Table:
+        """Returns a table that captures a snapshot of this whenever trigger_table updates.
+
+        When trigger_table updates, a snapshot of this table and the "stamp key" from trigger_table form the resulting
+        table. The "stamp key" is the last row of the trigger_table, limited by the stamp_cols. If {@code trigger}
+        is empty, the "stamp key" will be represented by {@code null} values.
+
+        Args:
+            trigger_table (Table): the trigger table
+            stamp_cols (Union[str, Sequence[str]): The stamp columns, may be renames.
+            initial (bool): Whether to take an initial snapshot upon construction, default is False. When False, the
+                resulting table will remain empty until trigger_table first updates.
+            incremental (bool): Whether the resulting table should be incremental, default is False. When True, only the
+                rows of this table that have been added or updated will have the latest "stamp key".
+            history (bool): Whether the resulting table should keep history, default is False. A history table appends a
+                full snapshot of this table and the "stamp key" as opposed to updating existing rows. Note: this flag is
+                currently incompatible with initial and incremental.
+
+        Returns:
+            a new table
+
+        Raises:
+            DHError
+        """
```

Comment:
```suggestion
        """Returns a table that captures a snapshot of this table whenever trigger_table updates.

        When trigger_table updates, a snapshot of this table and the "stamp key" from trigger_table form the resulting
        table. The "stamp key" is the last row of the trigger_table, limited by the stamp_cols. If {@code trigger}
        is empty, the "stamp key" will be represented by NULL values.

        Args:
            trigger_table (Table): the trigger table
            stamp_cols (Union[str, Sequence[str]): The stamp columns, may be renames.
            initial (bool): Whether to take an initial snapshot upon construction, default is False. When False, the
                resulting table will remain empty until trigger_table first updates.
            incremental (bool): Whether the resulting table should be incremental, default is False. When True, only the
                rows of this table that have been added or updated will have the latest "stamp key".
            history (bool): Whether the resulting table should keep history, default is False. A history table appends a
                full snapshot of this table and the "stamp key" as opposed to updating existing rows. Note: this flag is
                currently incompatible with initial and incremental.

        Returns:
            a new table

        Raises:
            DHError
        """
```

1) "If {@code trigger}
        is empty, the "stamp key" will be represented by NULL values." is not clear.
2) "{@code trigger}" looks like Javadoc that was copied over and is not a sphinx format we have used before.
3) "incremental (bool)" line could be made more clear.  There isn't anything else about incremental, so a new user is going to have a hard time with this description.
4) "Note: this flag is currently incompatible with initial and incremental." is not clear.  I would say that those values need to be false, since incompatible is vague in this context.
5) What happens if stamp_cols is empty / None?  How do you include no stamp cols?  How do you include all stamp cols?

Reply:
@jmao-denver Is `stamp_cols` or `cols` more consistent with the rest of the API?

Reply:
I actually prefer stamp_cols and the name isn't inconsistent.

Reply:
The API for snapshot_when is a little nuanced... I've updated them to hopefully make it clearer.